src/elements/OcModal.vue

```diff
@@ -213,23 +226,34 @@ export default {
     },
 
     $_ocModal_confirm() {
-      const value = this.hasInput ? this.$refs.ocModalInput.value : null
+      if (this.buttonConfirmDisabled || this.inputError) {
+        return
+      }
 
       /**
        * The user clicked on the confirm button. If input exists, emits it's value
        * 
-       * @property {String} inputValue Value of the input
+       * @property {String} value Value of the input
        */
-      this.$emit("confirm", value)
+      this.$emit("confirm", this.$_ocModal_input_value)
     },
 
-    $_ocModal_input_type() {
+    $_ocModal_input_type(value) {
+      // Assign value to the input value
+      this.$_ocModal_input_value = value
+
       /**
        * The user typed into the input
        * 
-       * @property {String} inputValue Value of the input
+       * @property {String} value Value of the input
        */
-      this.$emit("type", this.$refs.ocModalInput.value)
+      this.$emit("type", value)
+    },
+
+    $_ocModal_input_assignPropAsValue() {
+      if (this.inputValue) {
```

Comment:
Just thinking... is the `if` necessary? 🤔 What's the reason to disallow setting empty or null values on `this.$_ocModal_input_value`?

Reply:
My original thinking was not to do any assignments in case the prop has not been used since it's not a required one. But you're right, this would prevent using the prop to e.g. clear the input. Thinking now however if it makes sense then to check if `hasInput === true` not to assign the value even though the input is missing.